### PR TITLE
Migrate ChannelActions favorite and unfavorite channel to redux

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Migrate favorite and unfavorite channel actions to redux

Affected Areas:
* Mark and unmark channels as favorite

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12692

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: actions/channel_actions.jsx [hunk 1]
```diff
@@ -98,7 +98,7 @@ export function executeCommand(message, args, success, error) {
             const currentUserId = UserStore.getCurrentId();
             savePreferences(currentUserId, [{category, name, user_id: currentUserId, value: 'false'}])(dispatch, getState);
             if (ChannelUtils.isFavoriteChannel(channel)) {
-                unmarkFavorite(channel.id);
+                ChannelActions.unfavoriteChannel(channel.id);
```
Comment (saturninoabril) [inline actions/channel_actions.jsx]: Why not `dispatch(ChannelActions.unfavoriteChannel(channel.id));`?

File: actions/channel_actions.jsx [hunk 2]
```diff
@@ -364,7 +345,7 @@ export async function leaveChannel(channelId, success) {
 
     await ChannelActions.leaveChannel(channelId)(dispatch, getState);
     if (ChannelUtils.isFavoriteChannelId(channelId)) {
-        unmarkFavorite(channelId);
+        ChannelActions.unfavoriteChannel(channelId);
```
Comment (saturninoabril) [inline actions/channel_actions.jsx]: same here?

**Comment by enahum** [comment]
@saturninoabril changes made

File: actions/channel_actions.jsx [hunk 3]
```diff
@@ -355,6 +339,19 @@ export async function getChannelMembersForUserIds(channelId, userIds, success, e
     }
 }
 
+export async function leaveChannel(channelId, success) {
```
[the remark this answers is not in the document]
Comment (enahum) [reply actions/channel_actions.jsx]: Nice catch, probably a bad rebase

**Comment by enahum** [comment]
@sudheerDev removed the code that did not belong to this PR

**Comment by sudheerDev** [comment]
👍

**Comment by enahum** [comment]
This seems to be invalid after merging a PR that was based on this one

**Comment by DHaussermann** [comment]
Explored the behavior of this areas no issues found. 

This code has already been merged as part of PR#1898

**Pull request closed without merging**